src/Api/Controllers/CollectionsController.cs

```diff
@@ -411,8 +480,150 @@ private async Task<bool> CanViewCollectionAsync(Guid orgId, Guid collectionId)
         return false;
     }
 
+    [Obsolete("Pre-Flexible Collections logic. Will be replaced by CollectionsAuthorizationHandler.")]
     private async Task<bool> ViewAtLeastOneCollectionAsync(Guid orgId)
     {
+        DeprecatedPermissionsGuard();
+
         return await _currentContext.ViewAllCollections(orgId) || await _currentContext.ViewAssignedCollections(orgId);
     }
+
+    private async Task<CollectionResponseModel> Get_FC(Guid collectionId)
+    {
+        var collection = await _collectionRepository.GetByIdAsync(collectionId, _currentContext.UserId.Value);
+        var authorized = (await _authorizationService.AuthorizeAsync(User, collection, CollectionOperations.Read)).Succeeded;
+        if (!authorized)
+        {
+            throw new NotFoundException();
+        }
+
+        return new CollectionResponseModel(collection);
+    }
+
+    private async Task<CollectionAccessDetailsResponseModel> GetDetails_FC(Guid id)
+    {
+        // New flexible collections logic
+        var (collection, access) = await _collectionRepository.GetByIdWithAccessAsync(id);
+        var authorized = (await _authorizationService.AuthorizeAsync(User, collection, CollectionOperations.Read)).Succeeded;
+        if (!authorized)
+        {
+            throw new NotFoundException();
+        }
+
+        return new CollectionAccessDetailsResponseModel(collection, access.Groups, access.Users);
+    }
+
+    private async Task<ListResponseModel<CollectionAccessDetailsResponseModel>> GetManyWithDetails_FC(Guid orgId)
+    {
+        // We always need to know which collections the current user is assigned to
+        var assignedOrgCollections = await _collectionRepository
+            .GetManyByUserIdWithAccessAsync(_currentContext.UserId.Value, orgId);
+
+        var readAllAuthorized =
+            (await _authorizationService.AuthorizeAsync(User, null, CollectionOperations.ReadAll(orgId))).Succeeded;
```

Comment:
Thanks for pointing that out, I found a difference in their current authorization checks, `Get` allows users with `AccessImportExport` permission whereas `GetManyWithDetails` does not. Please re-check to be sure I didn't miss anything.

When I asked about different view models it was because some repository methods return classes that are different from the database model (usually from SQL views). So if we had an AuthorizationHandler using the database model as the type it wouldn't work, we would have to create another one.

Reply:
I'll dig into this more in my next review.